common/datavalues/src/types/data_type_coercion.rs

```diff
@@ -248,6 +248,7 @@ pub fn numerical_arithmetic_coercion(
             construct_numeric_type(true, has_float, next_size(max_size))
         }
         DataValueArithmeticOperator::Div => Ok(DataType::Float64),
+        DataValueArithmeticOperator::IntDiv => Ok(DataType::Float64),
```

Comment:
construct_numeric_type(has_signed, false,  sizeof (lhs)  )

Reply:
> `intDiv` return type should be integer?

The problem is with `N DIV 0` -> inf (f64)

Reply:
How about making N DIV 0 throw error: "Division by zero"  for intDiv.

Reply:
@sundy-li  It is absolutely legal to do `mysql -uroot -e 'SELECT 7.99 DIV 2.99'` -> 2 unlike `mysql -uroot -e 'SELECT 7 DIV 2'` -> 3
If we apply construct_numeric_type(has_signed, false, sizeof (...) ) 
- for case `[7] DIV [0]` - we get `Division by zero` error
- for case [7] DIV [0.0] - we got `[inf]`

If we want `Division by zero` error in any case the only way I can see is iterate through lhs and lookup for 0 and throw  `Division by zero` error
Please confirm.

Reply:
I implemented check divider for zero - pls see last commit in this pr